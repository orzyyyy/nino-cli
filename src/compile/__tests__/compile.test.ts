import { compile } from '..';
import fs from 'fs-extra';
import { joinWithRootPath } from '../../utils/common';

const outputDirPaths = ['lib', 'es'];
const outputFilePaths = [
  'ts/assets/case-1',
  'ts/assets/case-2',
  'ts/case-3',
  'ts/src/calc.js',
  'ts/src/assets/case-4',
  'ts/src/assets/case-5',
  'ts/src/modules/add/index.js',
  'ts/src/modules/add/css.css',
  'tsx/assets/case-1',
  'tsx/assets/case-2',
  'tsx/case-3-tsx',
  'tsx/index-tsx.js',
  'tsx/src/calc-tsx.js',
  'tsx/src/modules/add/index-tsx.js',
  'tsx/src/modules/add/css-tsx.css',
  'tsx/src/assets/case-4',
  'tsx/src/assets/case-5',
];

const getContent = (paths: string[]) => fs.readFileSync(joinWithRootPath(paths)).toString();

describe('nino compile', () => {
  it('compile correctly', done => {
    compile({ entry: 'cases/compile' }, () => {
      for (const dir of outputDirPaths) {
        for (const file of outputFilePaths) {
          jest.setTimeout(10 * 1000);
          expect(getContent([dir, file])).toMatchSnapshot();
        }
      }
      done();
    });
  });
});
