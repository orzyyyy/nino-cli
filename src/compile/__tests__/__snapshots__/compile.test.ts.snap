// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nino compile compile correctly 1`] = `
"case-1
"
`;

exports[`nino compile compile correctly 2`] = `
"case-2
"
`;

exports[`nino compile compile correctly 3`] = `
"case-3
"
`;

exports[`nino compile compile correctly 4`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.calc = void 0;

var calc = function calc() {
  return 'calc';
};

exports.calc = calc;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGMudHMiXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUFPLElBQU1BLElBQUksR0FBRyxTQUFQQSxJQUFPO0FBQUEsU0FBTSxNQUFOO0FBQUEsQ0FBYiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBjb25zdCBjYWxjID0gKCkgPT4gJ2NhbGMnO1xuIl19"
`;

exports[`nino compile compile correctly 5`] = `
"case-4
"
`;

exports[`nino compile compile correctly 6`] = `
"case-5
"
`;

exports[`nino compile compile correctly 7`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.add = void 0;

require(\\"./css.css\\");

var add = function add() {
  return 2;
};

exports.add = add;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LnRzIl0sIm5hbWVzIjpbImFkZCJdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUFBOztBQUVPLElBQU1BLEdBQUcsR0FBRyxTQUFOQSxHQUFNO0FBQUEsU0FBTSxDQUFOO0FBQUEsQ0FBWiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAnLi9jc3MuY3NzJztcblxuZXhwb3J0IGNvbnN0IGFkZCA9ICgpID0+IDI7XG4iXX0="
`;

exports[`nino compile compile correctly 8`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 9`] = `
"case-1
"
`;

exports[`nino compile compile correctly 10`] = `
"case-2
"
`;

exports[`nino compile compile correctly 11`] = `
"case-3-tsx
"
`;

exports[`nino compile compile correctly 12`] = `
"\\"use strict\\";

var _dateFns = require(\\"date-fns\\");

var _calcTsx = require(\\"./src/calc-tsx\\");

var defaultFunc = function defaultFunc(prefix) {
  console.log(\\"\\".concat(prefix, \\": \\").concat((0, _dateFns.format)(new Date(), 'yyyy-MM-dd hh:mm:ss'))); // eslint-disable-line

  console.log((0, _calcTsx.calc)()); // eslint-disable-line
};

defaultFunc('timeStamp:');
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LXRzeC50c3giXSwibmFtZXMiOlsiZGVmYXVsdEZ1bmMiLCJwcmVmaXgiLCJjb25zb2xlIiwibG9nIiwiRGF0ZSJdLCJtYXBwaW5ncyI6Ijs7QUFBQTs7QUFDQTs7QUFFQSxJQUFNQSxXQUFXLEdBQUcsU0FBZEEsV0FBYyxDQUFDQyxNQUFELEVBQW9CO0FBQ3RDO0FBQ0FDLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixXQUFlRixNQUFmLGVBQTBCLHFCQUFPLElBQUlHLElBQUosRUFBUCxFQUFtQixxQkFBbkIsQ0FBMUIsR0FGc0MsQ0FFa0M7QUFDeEU7O0FBQ0FGLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZLG9CQUFaLEVBSnNDLENBSWpCO0FBQ3RCLENBTEQ7O0FBT0FILFdBQVcsQ0FBQyxZQUFELENBQVgiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBmb3JtYXQgfSBmcm9tICdkYXRlLWZucyc7XG5pbXBvcnQgeyBjYWxjIH0gZnJvbSAnLi9zcmMvY2FsYy10c3gnO1xuXG5jb25zdCBkZWZhdWx0RnVuYyA9IChwcmVmaXg6IHN0cmluZykgPT4ge1xuICAvLyB0c2xpbnQ6ZGlzYWJsZS1uZXh0LWxpbmU6IG5vLWNvbnNvbGVcbiAgY29uc29sZS5sb2coYCR7cHJlZml4fTogJHtmb3JtYXQobmV3IERhdGUoKSwgJ3l5eXktTU0tZGQgaGg6bW06c3MnKX1gKTsgLy8gZXNsaW50LWRpc2FibGUtbGluZVxuICAvLyB0c2xpbnQ6ZGlzYWJsZS1uZXh0LWxpbmU6IG5vLWNvbnNvbGVcbiAgY29uc29sZS5sb2coY2FsYygpKTsgLy8gZXNsaW50LWRpc2FibGUtbGluZVxufTtcblxuZGVmYXVsdEZ1bmMoJ3RpbWVTdGFtcDonKTtcbiJdfQ=="
`;

exports[`nino compile compile correctly 13`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.calc = void 0;

var calc = function calc() {
  return 'calc';
};

exports.calc = calc;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGMtdHN4LnRzIl0sIm5hbWVzIjpbImNhbGMiXSwibWFwcGluZ3MiOiI7Ozs7Ozs7QUFBTyxJQUFNQSxJQUFJLEdBQUcsU0FBUEEsSUFBTztBQUFBLFNBQU0sTUFBTjtBQUFBLENBQWIiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgY29uc3QgY2FsYyA9ICgpID0+ICdjYWxjJztcbiJdfQ=="
`;

exports[`nino compile compile correctly 14`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.add = void 0;

require(\\"./css-tsx.css\\");

var add = function add() {
  return 2;
};

exports.add = add;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LXRzeC50c3giXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiOzs7Ozs7O0FBQUE7O0FBRU8sSUFBTUEsR0FBRyxHQUFHLFNBQU5BLEdBQU07QUFBQSxTQUFNLENBQU47QUFBQSxDQUFaIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICcuL2Nzcy10c3guY3NzJztcblxuZXhwb3J0IGNvbnN0IGFkZCA9ICgpID0+IDI7XG4iXX0="
`;

exports[`nino compile compile correctly 15`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 16`] = `
"case-4
"
`;

exports[`nino compile compile correctly 17`] = `
"case-5
"
`;

exports[`nino compile compile correctly 18`] = `
"case-1
"
`;

exports[`nino compile compile correctly 19`] = `
"case-2
"
`;

exports[`nino compile compile correctly 20`] = `
"case-3
"
`;

exports[`nino compile compile correctly 21`] = `
"export var calc = function calc() {
  return 'calc';
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGMudHMiXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxJQUFNQSxJQUFJLEdBQUcsU0FBUEEsSUFBTztBQUFBLFNBQU0sTUFBTjtBQUFBLENBQWIiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgY29uc3QgY2FsYyA9ICgpID0+ICdjYWxjJztcbiJdfQ=="
`;

exports[`nino compile compile correctly 22`] = `
"case-4
"
`;

exports[`nino compile compile correctly 23`] = `
"case-5
"
`;

exports[`nino compile compile correctly 24`] = `
"import './css.css';
export var add = function add() {
  return 2;
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LnRzIl0sIm5hbWVzIjpbImFkZCJdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxXQUFQO0FBRUEsT0FBTyxJQUFNQSxHQUFHLEdBQUcsU0FBTkEsR0FBTTtBQUFBLFNBQU0sQ0FBTjtBQUFBLENBQVoiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgJy4vY3NzLmNzcyc7XG5cbmV4cG9ydCBjb25zdCBhZGQgPSAoKSA9PiAyO1xuIl19"
`;

exports[`nino compile compile correctly 25`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 26`] = `
"case-1
"
`;

exports[`nino compile compile correctly 27`] = `
"case-2
"
`;

exports[`nino compile compile correctly 28`] = `
"case-3-tsx
"
`;

exports[`nino compile compile correctly 29`] = `
"import { format } from 'date-fns';
import { calc } from './src/calc-tsx';

var defaultFunc = function defaultFunc(prefix) {
  console.log(\\"\\".concat(prefix, \\": \\").concat(format(new Date(), 'yyyy-MM-dd hh:mm:ss'))); // eslint-disable-line

  console.log(calc()); // eslint-disable-line
};

defaultFunc('timeStamp:');
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LXRzeC50c3giXSwibmFtZXMiOlsiZm9ybWF0IiwiY2FsYyIsImRlZmF1bHRGdW5jIiwicHJlZml4IiwiY29uc29sZSIsImxvZyIsIkRhdGUiXSwibWFwcGluZ3MiOiJBQUFBLFNBQVNBLE1BQVQsUUFBdUIsVUFBdkI7QUFDQSxTQUFTQyxJQUFULFFBQXFCLGdCQUFyQjs7QUFFQSxJQUFNQyxXQUFXLEdBQUcsU0FBZEEsV0FBYyxDQUFDQyxNQUFELEVBQW9CO0FBQ3RDO0FBQ0FDLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixXQUFlRixNQUFmLGVBQTBCSCxNQUFNLENBQUMsSUFBSU0sSUFBSixFQUFELEVBQWEscUJBQWIsQ0FBaEMsR0FGc0MsQ0FFa0M7QUFDeEU7O0FBQ0FGLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZSixJQUFJLEVBQWhCLEVBSnNDLENBSWpCO0FBQ3RCLENBTEQ7O0FBT0FDLFdBQVcsQ0FBQyxZQUFELENBQVgiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBmb3JtYXQgfSBmcm9tICdkYXRlLWZucyc7XG5pbXBvcnQgeyBjYWxjIH0gZnJvbSAnLi9zcmMvY2FsYy10c3gnO1xuXG5jb25zdCBkZWZhdWx0RnVuYyA9IChwcmVmaXg6IHN0cmluZykgPT4ge1xuICAvLyB0c2xpbnQ6ZGlzYWJsZS1uZXh0LWxpbmU6IG5vLWNvbnNvbGVcbiAgY29uc29sZS5sb2coYCR7cHJlZml4fTogJHtmb3JtYXQobmV3IERhdGUoKSwgJ3l5eXktTU0tZGQgaGg6bW06c3MnKX1gKTsgLy8gZXNsaW50LWRpc2FibGUtbGluZVxuICAvLyB0c2xpbnQ6ZGlzYWJsZS1uZXh0LWxpbmU6IG5vLWNvbnNvbGVcbiAgY29uc29sZS5sb2coY2FsYygpKTsgLy8gZXNsaW50LWRpc2FibGUtbGluZVxufTtcblxuZGVmYXVsdEZ1bmMoJ3RpbWVTdGFtcDonKTtcbiJdfQ=="
`;

exports[`nino compile compile correctly 30`] = `
"export var calc = function calc() {
  return 'calc';
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGMtdHN4LnRzIl0sIm5hbWVzIjpbImNhbGMiXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sSUFBTUEsSUFBSSxHQUFHLFNBQVBBLElBQU87QUFBQSxTQUFNLE1BQU47QUFBQSxDQUFiIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IGNvbnN0IGNhbGMgPSAoKSA9PiAnY2FsYyc7XG4iXX0="
`;

exports[`nino compile compile correctly 31`] = `
"import './css-tsx.css';
export var add = function add() {
  return 2;
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluZGV4LXRzeC50c3giXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLGVBQVA7QUFFQSxPQUFPLElBQU1BLEdBQUcsR0FBRyxTQUFOQSxHQUFNO0FBQUEsU0FBTSxDQUFOO0FBQUEsQ0FBWiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAnLi9jc3MtdHN4LmNzcyc7XG5cbmV4cG9ydCBjb25zdCBhZGQgPSAoKSA9PiAyO1xuIl19"
`;

exports[`nino compile compile correctly 32`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 33`] = `
"case-4
"
`;

exports[`nino compile compile correctly 34`] = `
"case-5
"
`;
